pkg/templates/accesstoresources/template.go

```diff
@@ -0,0 +1,206 @@
+	rbacV1 "k8s.io/api/rbac/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "access-to-resources"
+)
+
+var (
+	roleGVK               = rbacV1.SchemeGroupVersion.WithKind(objectkinds.Role)
+	clusterRoleGVK        = rbacV1.SchemeGroupVersion.WithKind(objectkinds.ClusterRole)
+	roleBindingGVK        = rbacV1.SchemeGroupVersion.WithKind(objectkinds.RoleBinding)
+	clusterRoleBindingGVK = rbacV1.SchemeGroupVersion.WithKind(objectkinds.ClusterRoleBinding)
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Access to Resources",
+		Key:         templateKey,
+		Description: "Flag cluster role bindings and role bindings that grant access to the specified resource kinds and verbs",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.Role,
+				objectkinds.ClusterRole,
+				objectkinds.ClusterRoleBinding,
+				objectkinds.RoleBinding},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			resourceRegexes := make([]*regexp.Regexp, 0, len(p.Resources))
+			for _, res := range p.Resources {
+				r, err := regexp.Compile(res)
+				if err != nil {
+					return nil, errors.Wrapf(err, "invalid regex %s", res)
+				}
+				resourceRegexes = append(resourceRegexes, r)
+			}
+			verbRegexes := make([]*regexp.Regexp, 0, len(p.Verbs))
+			for _, verb := range p.Verbs {
+				v, err := regexp.Compile(verb)
+				if err != nil {
+					return nil, errors.Wrapf(err, "invalid regex %s", verb)
+				}
+				verbRegexes = append(verbRegexes, v)
+			}
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				bindingGVK := extract.GVK(object.K8sObject)
+				if bindingGVK == roleBindingGVK {
+					binding, ok := object.K8sObject.(*rbacV1.RoleBinding)
+					if !ok {
+						return nil
+					}
+					namespace := stringutils.OrDefault(binding.Namespace, "default")
+					return findRole(binding.RoleRef.Name, namespace, lintCtx, resourceRegexes, verbRegexes, p.FlagRolesNotFound)
+				}
+
+				if bindingGVK == clusterRoleBindingGVK {
+					binding, ok := object.K8sObject.(*rbacV1.ClusterRoleBinding)
+					if !ok {
+						return nil
+					}
+					return findClusterRole(binding.RoleRef.Name, lintCtx, resourceRegexes, verbRegexes, p.FlagRolesNotFound)
+				}
+				return nil
+			}, nil
+		}),
+	})
+}
+
+// find clusterrole by name, and check if it has access to the specified resource kinds and verbs
+func findClusterRole(name string, lintCtx lintcontext.LintContext, resourceRegexes, verbRegexes []*regexp.Regexp, flag bool) []diagnostic.Diagnostic {
+	results := []diagnostic.Diagnostic{}
+	clusterroles := []*rbacV1.ClusterRole{}
+	for _, object := range lintCtx.Objects() {
+		gvk := extract.GVK(object.K8sObject)
+		if gvk != clusterRoleGVK {
+			continue
+		}
```

Comment:
Again, you can remove this I think. It's redundant with the below.

Reply:
yes, removed